docs/_guides/nifi-jms-jndi.md

```diff
@@ -0,0 +1,695 @@
+---
+layout: guides
+title: JNDI Connection Factory Provider for Apache NiFi
+summary: Apache NiFi is a software project from the Apache Software Foundation which enables the automation of data flow between systems. It can be described as data logistics. Similar to how parcel services move and track packages, Apache NiFi helps move and track data. 
+icon: nifilogo.png
+links:
+    - label: Sample Source Code - NiFi JMS JNDI Connection Service
+      link: https://github.com/SolaceLabs/solace-integration-guides/blob/master/src/nifi-jms-jndi
+---
+
+## Overview
+
+This document demonstrates how to integrate the Solace Java Message Service (JMS) with Apache NiFi JMS Processor for JMS message publish and consumption. The goal of this document is to outline best practices for this integration to enable efficient use of both Apache NiFi and Solace JMS. 
+
+![Sequence Diagram]({{ site.baseurl }}/images/nifi-jms-jndi/nifisequence.png)
+
+The target audience of this document is developers with knowledge of both NiFi and JMS in general. As such this document focuses on the technical steps required to achieve the integration. For detailed background on either Solace JMS or NiFi refer to the referenced documents below.
+
+This document is divided into the following sections to cover the Solace JMS integration with Apache NiFi:
+
+* Integrating Solace with NiFi JMS Processor
+* Working with Solace High Availability
+* Debugging Tips 
+
+
+### Related Documentation
+
+These links contain information related to this guide:
+
+* [Solace Developer Portal]({{ site.links-dev-portal }}){:target="_top"}
+* [Solace Messaging API for JMS]({{ site.links-docs-jms }}){:target="_top"}
+* [Solace JMS API Online Reference Documentation]({{ site.links-docs-jms-api }}){:target="_top"}
+* [Solace Feature Guide]({{ site.links-docs-features }}){:target="_top"}
+* [Solace Message Router Configuration]({{ site.links-docs-router-config }}){:target="_top"}
+* [Solace Command Line Interface Reference]({{ site.links-docs-cli }}){:target="_top"}
+* [NiFi Documentation](http://nifi.apache.org/docs.html){:target="_blank"}
+* [NiFi Developer Guide](http://nifi.apache.org/developer-guide.html){:target="_blank"}
+
+## Integrating Solace with NiFi JMS Processor
+
+NiFi JMS Bundle comes with 2 processors, ConsumeJMS and PublishJMS. The 2 out-of-box processors work well with some messaging brokers, i.e. ActiveMQ and etc. However, not all messaging brokers work well with the built-in JMS processors that expect zero-argument connection factory constructors. Solace connection factory and some other JMS connection factories cannot be instantiated with zero argument. There is a current NiFi Jira on [this major bug](https://issues.apache.org/jira/browse/NIFI-2701){:target="_blank"}.
```

Comment:
Solace has a way to work with zero-argument connection factor constructors:

SolConnectionFactory cf = SolJmsUtility.createConnectionFactory();
cf.setHost((String) args[0]);
cf.setVPN("default");
cf.setUsername("clientUsername");
cf.setPassword("password");

From http://dev.solace.com/get-started/jms-tutorials/publish-subscribe_jms/

Could that be used to help out?